src/commons/api/roas.rs

```diff
@@ -129,29 +129,35 @@ impl RoaAggregateKeyFmtError {
     }
 }
 
-//------------ RoaDefinition -----------------------------------------------
+//------------ RoaPayload --------------------------------------------------
 
-/// This type defines the definition of a Route Origin Authorization (ROA), i.e.
-/// the originating asn, IPv4 or IPv6 prefix, and optionally a max length.
+/// This type defines the definition of a Route Origin Authorization (ROA)
+/// payload: ASN, Prefix and optional Max Length
+///
+/// Note that a ROA object may contain multiple payloads (RoaDefinitions)
+/// aggregated by (the same) ASN. This is discouraged and Krill prefers
+/// to generate a dedicated ROA object for each payload, but aggregation
+/// will be done when a (configurable) threshold is exceeded.
 #[derive(Clone, Copy, Deserialize, Eq, Hash, PartialEq, Serialize)]
-pub struct RoaDefinition {
+pub struct RoaPayload {
     asn: AsNumber,
     prefix: TypedPrefix,
     #[serde(skip_serializing_if = "Option::is_none")]
     max_length: Option<u8>,
 }
 
-impl RoaDefinition {
+impl RoaPayload {
     pub fn new(asn: AsNumber, prefix: TypedPrefix, max_length: Option<u8>) -> Self {
-        RoaDefinition {
+        RoaPayload {
             asn,
             prefix,
             max_length,
         }
     }
 
-    pub fn explicit_max_length(self) -> Self {
-        RoaDefinition {
+    /// Ensures that the payload uses an explicit max length
+    pub fn into_explicit_max_length(self) -> Self {
```

Comment:
On `RoaConfigurationUpdates` this is called `into_explicit`. Should these match? Perhaps the longer more descriptive name should be used in both places?

Reply:
using the longer `into_explicit_max_length` now.